Add support for multiple controller instances

greenboxal opened the pull request:
This allows multiple instances of kube-aws-ingress-controller to coexist in the same cluster.

It allows to properly split different types of trafic or routers (using Skipper and another router) at the same time.

greenboxal commented on aws/ec2.go:
```diff
@@ -14,6 +14,7 @@ const (
 	kubernetesClusterLegacyTag = "KubernetesCluster"
 	clusterIDTag               = "ClusterID" // TODO(sszuecs): deprecated fallback cleanup
 	clusterIDTagPrefix         = "kubernetes.io/cluster/"
+	controllerIDTagPrefix      = "zalando.com/ingress-aws-controller/"
```
Fixed

greenboxal commented:
@mikkeloscar addressed your comments. I also improved the handling of the default controller ID so the variables/consts/default are more natural.

mikkeloscar commented:
:+1:

hjacobs commented:
What happens during migration from a previous version? Would all stacks be recreated or updated?

mikkeloscar commented:
@hjacobs if you don't specify the flag nothing will happen as the default is the same as before.

If you specify the flag with a different value from the default then your existing stacks will no longer be managed by the controller and it will instead create new stacks. The existing stacks would be orphaned in that case because no controller is managing them.

szuecs commented:
@greenboxal can you add an entry in the docs please for your nice feature, which people want to use?
Thanks for your work, good code and also a test was added!
Please also show how to use it. :)

greenboxal commented:
Sorry for not doing this earlier, just addressed your comments:

* Added section to README
* Removed backward compatibility and added note to README
  - I didn't add to the top section about upgrades because I don't know which one to use. Are we releasing a new minor/major with this?

mikkeloscar commented:
:+1:

hjacobs commented:
:+1:
